### PR TITLE
Max connections ocaml side

### PR DESCRIPTION
**Pull request by imeckler**
This PR is based on https://github.com/MinaProtocol/mina/pull/7725/. It adds libp2p fixes:

- get rid of the trimming grace period
- call trim when appropriate

and then adds the flag for enabling @noot's peer exchange to the networking config.

### CODE REVIEW

File: src/app/libp2p_helper/src/codanet.go [hunk 1]
```diff
@@ -454,5 +601,16 @@ func MakeHelper(ctx context.Context, listenOn []ma.Multiaddr, externalAddr ma.Mu
 		GatingState:       gatingState,
 		ConnectionManager: connManager,
 		BandwidthCounter:  bandwidthCounter,
-	}, nil
+		Seeds:             seeds,
```
Comment (imeckler) [inline src/app/libp2p_helper/src/codanet.go:604]: Is this being set properly on OCaml side?

File: src/app/libp2p_helper/src/codanet.go [hunk 2]
```diff
@@ -333,46 +420,102 @@ func (gs *CodaGatingState) InterceptUpgraded(network.Conn) (allow bool, reason c
...
+			_ = s.Close()
+			return
+		} else if err != nil {
+			logger.Errorf("failed to decode list of peers err=%s", err)
+			continue
+		}
+
+		for _, peer := range peers {
+			go func() {
+				err = h.Host.Connect(h.Ctx, peer)
```
Comment (bkase) [inline src/app/libp2p_helper/src/codanet.go:498]: Does this properly add to our peer store for further gossiping?

File: src/app/libp2p_helper/src/codanet.go (hunk 1 above)
Comment (imeckler) [reply src/app/libp2p_helper/src/codanet.go:604]: Change add_peer to add_seed_peer

**Review by bkase — COMMENTED** [review]
We'll need to treat added peers as seed peers to get this whole system working properly with dynamically added peer lists

File: src/app/libp2p_helper/src/codanet.go (hunk 2 above)
Comment (bkase) [reply src/app/libp2p_helper/src/codanet.go:498]: Please leave a TODO for now

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Comment by mrmr1993** [comment]
Something went wrong with code review here. This PR changed the graphql schema in a non-backwards-compatible way, breaking the `add-peers` command (see #7838). That change appears not to have been reviewed -- more broadly, this PR was merged without being explicitly approved here by anyone -- and we've now shipped a bug.

**Pull request merged**